### PR TITLE
Improving Support for OSX & Rider

### PR DESCRIPTION
**Pull request by joshgarnett**
* Rider/OSX appears to be less forgiving around the csproj format and requires that the nuget dependencies use Version and not version.
* Updated the build.sh script to only install dotnet once and also to pull down protoc from nuget.
* The FAKE build has been updated to support building the Protobuf target on OSX & Linux

### CODE REVIEW

**Comment by Arkatufus** [comment]
Looks good :smile:

**Comment by Aaronontheweb** [comment]
I'm not sure what the issue is here exactly, but the Mono test runner is hitting the deck when it starts scanning for test assemblies to run... What could be causing that?

**Comment by joshgarnett** [comment]
It doesn't seem like the failure was related to the change from what I can tell.  I'll look at getting the branch caught up to dev.

File: build.sh [hunk 1]
```diff
@@ -42,17 +44,19 @@ fi
 # INSTALL .NET CORE CLI
 ###########################################################################
 
-echo "Installing .NET CLI..."
```
Comment (Danthar) [inline build.sh]: This change for not installing .NET CLI is whats causing the issue here. Revert this section and it should solve your problem

File: build.sh [hunk 2]
```diff
@@ -87,6 +91,30 @@ if [ ! -f "$FAKE_EXE" ]; then
 fi
 
 ###########################################################################
+# INSTALL PROTOC
+###########################################################################
+
+unameOut="$(uname -s)"
```
Comment (Danthar) [inline build.sh]: What does this line do ?

**Pull request merged**